lib/logstash/filters/util/lookup.rb

```diff
@@ -0,0 +1,156 @@
+require_relative "lookup_result"
+
+module LogStash module Filters module Util
+  class Lookup
+    class Sprintfier
+      def initialize(param)
+        @param = param
+      end
+
+      def fetch(event)
+        event.sprintf(@param)
+      end
+    end
+
+    class Getfier
+      def initialize(param)
+        @param = param
+      end
+
+      def fetch(event)
+        event.get(@param)
+      end
+    end
+
+    attr_reader :target, :query, :parameters
+
+    def self.validate(array_of_options)
+      errors = []
+      array_of_options.each_with_index do |options, i|
+        instance = new(options, {}, Object.new, "lookup-#{i.next}")
+        unless instance.valid?
+          errors << instance.formatted_errors
+        end
+      end
+      return nil if errors.empty?
+      errors.join("; ")
+    end
+
+    def initialize(options, globals, logger, default_id)
+      @target = options["target"]
+      @id = target || default_id
+      @options = options
+      @globals = globals
+      @logger = logger
```

Comment:
True, It will have context about the support class but we will lose it association with the plugin class.